integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/nextmove/ServiceBusRestClient.java

```diff
@@ -0,0 +1,181 @@
+    }
+
+    public void sendMessage(byte[] message, String queuePath) {
+        String resourceUri = format("https://%s.servicebus.windows.net/%s/messages",
+                props.getNextbest().getServiceBus().getNamespace(), queuePath);
+        URI uri = convertToUri(resourceUri);
+
+        String auth = createAuthorizationHeader(resourceUri);
+        HttpHeaders headers = new HttpHeaders();
+        headers.add("Authorization", auth);
+        headers.add("BrokerProperties", "{}");
+        HttpEntity<Object> httpEntity = new HttpEntity<>(message, headers);
+
+        RestTemplate restTemplate = new RestTemplate();
+        ResponseEntity<String> response = restTemplate.exchange(uri, HttpMethod.POST, httpEntity, String.class);
+        if (!response.getStatusCode().is2xxSuccessful()) {
+            log.error("{} got response {}, error sending message to service bus", resourceUri, response.getStatusCode().toString());
+        }
+    }
+
+    public Optional<ServiceBusMessage> receiveMessage() {
+        String resourceUri = format("https://%s.servicebus.windows.net/%s/messages/head",
+                props.getNextbest().getServiceBus().getNamespace(),
+                localQueuePath);
+
+        String auth = createAuthorizationHeader(resourceUri);
+        HttpHeaders headers = new HttpHeaders();
+        headers.add("Authorization", auth);
+        HttpEntity<Object> httpEntity = new HttpEntity<>(headers);
+
+        URI uri = convertToUri(resourceUri);
+        RestTemplate restTemplate = new RestTemplate();
+        ResponseEntity<String> response = restTemplate.exchange(uri, HttpMethod.POST, httpEntity, String.class);
+        if (!asList(HttpStatus.OK, HttpStatus.CREATED).contains(response.getStatusCode())) {
+            log.debug("{} got response {}, returning empty", resourceUri, response.getStatusCode().toString());
+            return Optional.empty();
+        }
+
+        ServiceBusMessage.ServiceBusMessageBuilder sbmBuilder = ServiceBusMessage.builder();
+        String brokerPropertiesJson = response.getHeaders().getFirst("BrokerProperties");
+        JsonParser jsonParser = new JsonParser();
+        JsonObject brokerProperties = jsonParser.parse(brokerPropertiesJson).getAsJsonObject();
+        sbmBuilder.lockToken(brokerProperties.get("LockToken").getAsString())
+                .messageId(brokerProperties.get("MessageId").getAsString())
+                .sequenceNumber(brokerProperties.get("SequenceNumber").getAsString());
+
+        try {
+            Unmarshaller unmarshaller = jaxbContext.createUnmarshaller();
+            EduDocument eduDocument = unmarshaller.unmarshal(new StringSource(response.getBody()), EduDocument.class).getValue();
+            sbmBuilder.body(eduDocument);
+            log.debug(format("Received message on queue=%s with conversationId=%s", localQueuePath, eduDocument.getConversationId()));
+        } catch (JAXBException e) {
+            e.printStackTrace();
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Use a logger to log this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1148)